src/core/CheckBox/base.ts

```diff
@@ -0,0 +1,3 @@
+import { CheckboxProps } from '@material-ui/core/Checkbox';
+
+export type BaseCheckboxProps = Omit<CheckboxProps, 'checked'>;
```

Comment:
Here `checked` is Omitted but again declared as a type in `checkbox.tsx`. If we need `checked` there is no need to Omit it. 

This `base.ts` file is not required.